src/os/include/IMutex.h

```diff
@@ -0,0 +1,25 @@
+#ifndef __IMUTEX_H_
+#define __IMUTEX_H_
+
+#include <mutex>
+
+/**
+ *@brief A class to manage mutual exclusion for guarding resources (uses static allocation for FreeRTOS)*/
```

Comment:
The static allocation comments could probably go in the implementation instead of the interface